#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From ac63ef3a9f13e808e4ed573499e43cd00d058f64 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Fri, 15 Mar 2024 11:31:03 +0100
Subject: [PATCH 1/2] WIP: When crosscompiling with Qt6 look for host moc

--
This should be fixed differently by adding host builds to
Gpg4win and migt vary from distro.
---
 m4/qt6.m4 | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/m4/qt6.m4 b/m4/qt6.m4
index dcb0313..65771f5 100644
--- a/m4/qt6.m4
+++ b/m4/qt6.m4
@@ -27,11 +27,22 @@ AC_DEFUN([FIND_QT6],
     # Qt6 moved moc to libexec
     qt6libexecdir=$($PKG_CONFIG --variable=libexecdir 'Qt6Core >= 6.4.0')
     AC_PATH_TOOL(MOC6, moc, [], [$qt6libexecdir])
+    if test -z "$MOC6"; then
+      if test "$have_w32_system" = "yes"; then
+        AC_PATH_TOOL(MOC6, moc, [], [/usr/libexec/qt6])
+      fi
+    fi
     if test -z "$MOC6"; then
       AC_MSG_WARN([moc not found - Qt 6 binding will not be built.])
       have_qt6_libs="no";
     fi
+
     AC_PATH_TOOL(RCC6, rcc, [], [$qt6libexecdir])
+    if test -z "$RCC6"; then
+      if test "$have_w32_system" = "yes"; then
+        AC_PATH_TOOL(RCC6, rcc, [], [/usr/libexec/qt6])
+      fi
+    fi
     if test -z "$RCC6"; then
       AC_MSG_WARN([rcc not found - Qt 6 binding will not be built.])
       have_qt6_libs="no";
--
2.45.1
